### PR TITLE
feat: use tanstack table

### PR DESCRIPTION
**Pull request by matalex412**
- add tanstack table component
- add pagination to tanstack table

image
*Tanstack table with pagination*

### CODE REVIEW

File: components/ApplicantTable.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  columnHelper.accessor('firstName', {
+    cell: (info) => info.getValue(),
+  })
+]
+
+const ApplicantTable = ({ applicants }: { applicants: ApplicantRow[] }) => {
+    return (
+        <TanstackTable data={applicants} columns={columns} />
+    )
+}
```
Comment (procaconsul) [inline components/ApplicantTable.tsx:23]: I would extract the props interface and use the parameterized FC type for the component (in general, I would stick to this style across the codebase). Remember to import FC. Also, the return of the functional component can be implicit when there's no other logic involved.

```suggestion

interface ApplicantTableProps {
    applicants: ApplicantRow[]
}

const ApplicantTable: FC<ApplicantTableProps>  = ({ applicants }) => (
    <TanstackTable data={applicants} columns={columns} />
)
```

File: components/TanstackTable/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import styles from './pagination.module.css'
+
+
+interface PaginationProps {
+  page: number
+  setPage: (page: number) => void
+  totalPages: number
+}
+
+const Pagination = ({ page, setPage, totalPages }: PaginationProps) => {
```
Comment (procaconsul) [inline components/TanstackTable/Pagination.tsx:16]:
```suggestion
const Pagination: FC<PaginationProps> = ({ page, setPage, totalPages }) => {
```

File: components/TanstackTable/pagination.module.css
Comment (procaconsul) [inline components/TanstackTable/pagination.module.css:1]: Can we not use tailwind classes instead of adding a custom css file?

**Review by procaconsul — CHANGES_REQUESTED** [review]
General remark: I noticed that some of the files use single quotes, some others use double quotes. My guess is that prettier wasn't run consistently on these files. Are husky pre-commit scripts kicking in for you on every commit?

File: components/ApplicantTable.tsx (hunk 1 above)
Comment (matalex412) [reply components/ApplicantTable.tsx]: Resolved in [8918a16](https://github.com/imperial/ug-admissions/pull/2/commits/8918a16e73a190d6a940009a0e43c813bd3f7e9f)

File: components/TanstackTable/Pagination.tsx (hunk 2 above)
Comment (matalex412) [reply components/TanstackTable/Pagination.tsx]: Resolved in [5c5505a](https://github.com/imperial/ug-admissions/pull/2/commits/5c5505a10711d439db85b5816413685733d6c0b8)

File: components/TanstackTable/pagination.module.css
Comment (matalex412) [reply components/TanstackTable/pagination.module.css:1]: Yes was having some issues with tailwind but sorted today in [068da89](https://github.com/imperial/ug-admissions/pull/2/commits/068da89cfb3b51c522398d1c5a44f554f3213f01)

**Comment by matalex412** [comment]
> General remark: I noticed that some of the files use single quotes, some others use double quotes. My guess is that prettier wasn't run consistently on these files. Are husky pre-commit scripts kicking in for you on every commit?

We think this is fixed now. Prettier is formatting imports to use single quotes and double quotes elsewhere, but presumably that's fine as long as it's consistent.

**Review by procaconsul — APPROVED** [review]

**Pull request merged**